src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/CrashHelmetListener.java

```diff
@@ -0,0 +1,34 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners;
+
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+
+public class CrashHelmetListener implements Listener {
+
+    public CrashHelmetListener(@Nonnull SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onPlayerCrash(EntityDamageEvent e) {
+        if (!(e.getEntity() instanceof Player)) return;
+        if (!(e.getCause() == EntityDamageEvent.DamageCause.FALL ||
+            e.getCause() == EntityDamageEvent.DamageCause.FLY_INTO_WALL)) return;
+
+        Player p = (Player) e.getEntity();
+        if (p.isGliding()) {
+            ItemStack stack = p.getInventory().getHelmet();
+            SlimefunItem item = SlimefunItem.getByItem(stack);
+            if ((item != null) && (item.getID().equals("CRASH_HELMET"))) {
+                e.setDamage(0);
```

Comment:
Setting the damage is not really good practice, you should rather cancel the event instead (and maybe play a sound effect too?).

Reply:
i tried cancelling... didnt really work for some reason

Reply:
also what sound effect would u suggest? id think either smth do do with iron or slime

Reply:
I'd try stepping or breaking bedrock or obsidian noise.

Reply:
confirmed: cancelling event you still die from falling form a high place

Reply:
> confirmed: cancelling event you still die from falling form a high place

Afaik cancelling death event doesn't cancel the death but only cancels the item drops and etc. This is a bug/feature of Spigot.

Reply:
if you dont die you still take damage

Reply:
> if you dont die you still take damage

From what I can remember you need to listen to player damage event. It's fired before player dies.

Reply:
> 
> Afaik cancelling death event doesn't cancel the death but only cancels the item drops and etc. This is a bug/feature of Spigot.

I meant cancelling the damage event